pkg/controllers/status/cluster_status_controller.go

```diff
@@ -60,8 +60,10 @@ type ClusterStatusController struct {
 	PredicateFunc               predicate.Predicate
 	InformerManager             informermanager.MultiClusterInformerManager
 	StopChan                    <-chan struct{}
-	ClusterClientSetFunc        func(c *v1alpha1.Cluster, client client.Client) (*util.ClusterClient, error)
+	ClusterClientSetFunc        func(*v1alpha1.Cluster, client.Client, util.ClientOption) (*util.ClusterClient, error)
```

Comment:
What about using `*util.ClientOption` to avoid copy?

Reply:
Sounds good.